.github/workflows/build-push-docker-images.yaml

```diff
@@ -0,0 +1,183 @@
+name: Build-Push-Docker-Image
+
+on:
+  push:
+    branches:
+      - build-push-docker-images
+
+jobs:
+  build-push-base-rvt-image:
+
+    name: Build and push RVT base Docker image
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        include:
+          - username: rvt
+            user_uid: 1000
```

Comment:
Does this depend on the user_uid/gid matching the user's uid/gid?
On my personal laptop this is true: I am 1000/1000.
But on my work laptop, I have different uid/gid.

My concern with using this uid/gid is that when I use the existing docker image, invoking cargo-verify or invoking KLEE directly creates files in the host filesystem with whatever uid/gid the docker image uses.
If those don't match the uid/gid in the host machine (eg when on my work laptop), the files have invalid uid/gid values and I couldn't even delete them.
Our solution was to copy the uid/gid from the host account into docker image.

I think that our current solution to the problem is wrong (it breaks the whole idea of shared images on docker hub).
But I don't understand Docker well enough to know whether this works either.

Reply:
As the user uid and user gid are declared in this section, the built image might retain this numbers.
I believe it could be an issue in the case you've described
(with another machine, different uid / gid and a mounted directory).
However there seems to be a way how to remap user uid / gid :
https://docs.docker.com/engine/security/userns-remap/#about-remapping-and-subordinate-user-and-group-ids

I have to confess that I've never needed to use remapping as I usually don't modify the files mounted in containers as a host user (and when I do I try to have a host user and container user belonging to a group having the same gid).